mls-rs-provider-sqlite/src/lib.rs

```diff
@@ -69,9 +95,11 @@ where
 {
     pub fn new(
         connection_strategy: CS,
+        journal_mode: Option<JournalMode>,
```

Comment:
Have you considered not adding this parameter and instead having a separate function like the following?

```rust
/// A `journal_mode` of `None` means the SQLite default is used.
pub fn with_journal_mode(self, journal_mode: Option<JournalMode>) -> Self {
    Self {
        journal_mode,
        ..self
    }
}
```

This has the advantage of not being a breaking change. The current approach is a breaking change for `mls-rs-provider-sqlite` and `mls-rs`.